huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_domain_test.go

```diff
@@ -0,0 +1,207 @@
+					resource.TestCheckResourceAttr(resourceName, "server.0.port", "8080"),
+					resource.TestCheckResourceAttrSet(resourceName, "policy_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "cname"),
+					resource.TestCheckResourceAttrSet(resourceName, "sub_domain"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckWafDomainV1Destroy(s *terraform.State) error {
+	config := acceptance.TestAccProvider.Meta().(*config.Config)
+	wafClient, err := config.WafV1Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return fmt.Errorf("error creating HuaweiCloud WAF client: %s", err)
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "huaweicloud_waf_domain" {
+			continue
+		}
+
+		_, err := domains.Get(wafClient, rs.Primary.ID).Extract()
+		if err == nil {
+			return fmt.Errorf("Waf domain still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckWafDomainV1Exists(n string, domain *domains.Domain) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := acceptance.TestAccProvider.Meta().(*config.Config)
+		wafClient, err := config.WafV1Client(acceptance.HW_REGION_NAME)
+		if err != nil {
+			return fmt.Errorf("error creating HuaweiCloud WAF client: %s", err)
+		}
+
+		found, err := domains.Get(wafClient, rs.Primary.ID).Extract()
+		if err != nil {
+			return err
+		}
+
+		if found.Id != rs.Primary.ID {
+			return fmt.Errorf("Waf domain not found")
+		}
+
+		*domain = *found
+
+		return nil
+	}
+}
+
+func testAccWafDomainV1_basic(certificateName string, name string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_waf_domain" "domain_1" {
+  domain           = "www.%s.com"
+  certificate_id   = huaweicloud_waf_certificate.certificate_1.id
+  certificate_name = huaweicloud_waf_certificate.certificate_1.name
+  keep_policy      = false
+  proxy            = false
+
+  server {
+    client_protocol = "HTTPS"
+    server_protocol = "HTTP"
+    address         = "119.8.0.14"
+    port            = 8080
+  }
+}
+`, testAccWafCertificateV1_conf(name), name)
+}
+
+func testAccWafDomainV1_update(certificateName string, name string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_waf_domain" "domain_1" {
+  domain           = "www.%s.com"
+  certificate_id   = huaweicloud_waf_certificate.certificate_1.id
+  certificate_name = huaweicloud_waf_certificate.certificate_1.name
+  keep_policy      = false
+  proxy            = true
+
+  server {
+    client_protocol = "HTTPS"
+    server_protocol = "HTTP"
+    address         = "119.8.0.14"
+    port            = 8443
+  }
+
+}
+`, testAccWafCertificateV1_conf(name), name)
+}
+
+func testAccWafDomainV1_policy(certificateName string, name string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_waf_policy" "policy_1" {
```

Comment:
seems `huaweicloud_waf_policy` has not defined

Reply:
---
One of domain unit testing depends on the policy. It has successfully run on my local. After adding the policy function, then it can be run on the server.
---
suggest to delete the acceptance test in this PR, and add it back when you finish `huaweicloud_waf_policy`.

Reply:
has deleted